### PR TITLE
Fix issue when updating a free access plan

### LINKED ISSUE #267: Request to `access-plans/{id}` results in an error when no `price` is supplied for free access plans
### Reproduction Steps

+ Make an API request to update any field of an access plan, for example the plan's `title` property.
+ Exclude the `price` property from the body of the API request.

Example payload:

```json
{"title":"An updated title"}
```

### Expected Behavior

+ The title of the plan will be updated.

### Actual Behavior

+ A 500 response is received from the API request

### Error Messages / Logs

```json
{
    "code": "invalid_meta",
    "message": "Cannot insert/update the price meta",
    "data": null
}

```

### Notes

I encountered this while developing the react-powered UI for managing access plans. This is easy enough to work around but ideally we shouldn't have to provide the price with every update request.

### On closer inspection

This problem arises when not supplying the price for a free access plan or when supplying the existing price on paid access plans.

EG: If I leave the price empty for a free plan (by excluding the key from the payload) I get this error.

With a paid plan, if I specify the existing plan amount I also get this error.

This leads me to believe that this is actually an issue we've previously identified with regards to how we return `false` from the core postmodel class when updating meta data (and the value is unchanged). This is treated as an error by the rest api since there's no way to disambiguate between a false response that means NO CHANGE MADE and false response resulting from an error.

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
## Description

Fixes #267 

## How has this been tested?
manually and with new unit tests

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: includes/server/class-llms-rest-access-plans-controller.php [hunk 1]
```diff
@@ -636,7 +637,7 @@ private function add_subordinate_props( &$to_set, $saved_props, $creating ) {
 		} else {
 
 			$subordinate_props['is_free']     = 'yes';
-			$subordinate_props['price']       = 0;
+			$subordinate_props['price']       = (float) 0;
```
Comment (pondermatic) [inline includes/server/class-llms-rest-access-plans-controller.php]: An alternative to casting. Either way works though.
```suggestion
			$subordinate_props['price']       = 0.0;
```

**Comment by pondermatic** [comment]
While reviewing this, I found a typo in [`LLMS_REST_Access_Plans_Controller::update_additional_object_fields()`](https://github.com/gocodebox/lifterlms-rest/blob/trunk/includes/server/class-llms-rest-access-plans-controller.php#L565) that is likely causing a bug. Should I open up a new issue?

**Review by pondermatic — DISMISSED** [review]

**Comment by eri-trabiccolo** [comment]
> LLMS_REST_Access_Plans_Controller::update_additional_object_fields()

Thanks @pondermatic.
I'm fixing it in the same PR :D

File: includes/server/class-llms-rest-access-plans-controller.php (hunk 1 above)
Comment (eri-trabiccolo) [reply includes/server/class-llms-rest-access-plans-controller.php]: yeah I guess your way is better <3

**Comment by eri-trabiccolo** [comment]
@pondermatic can you give another look please?

**Review by pondermatic — APPROVED** [review]

**Pull request merged**